onnxruntime/core/providers/shared_library/bridge.h

```diff
@@ -0,0 +1,95 @@
+namespace onnxruntime {
+
+struct IExecutionProviderFactory;
+struct ProviderHost;
+struct KernelCreateInfo;
+class CPUIDInfo;
+
+namespace logging {
+class Logger;
+}
+
+struct Provider {
+  virtual std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory(int device_id) = 0;
+  virtual void SetProviderHost(ProviderHost& host) = 0;
+};
+
+// There are two ways to route a function, one is a virtual method and the other is a function pointer (or pointer to member function)
+// The function pointers are nicer in that they directly call the target function, but they cannot be used in cases where we're calling
+// a specific implementation of a virtual class member. Trying to get a pointer to member of a virtual function will return a thunk that
+// calls the virtual function (which will lead to infinite recursion in the bridge). There is no known way to get the non virtual member
+// function pointer implementation in this case.
+struct ProviderHost {
+  virtual logging::Logger* LoggingManager_GetDefaultLogger() = 0;
+
+  MLDataType (*DataTypeImpl_GetType_Tensor)();
+  MLDataType (*DataTypeImpl_GetType_float)();
+  MLDataType (*DataTypeImpl_GetTensorType_float)();
+
+  virtual void* HeapAllocate(size_t size) = 0;
+  virtual void HeapFree(void*) = 0;
+
+  const ::std::string& (*google_protobuf_internal_GetEmptyStringAlreadyInited)();
+  void (google::protobuf::internal::RepeatedPtrFieldBase::*google_protobuf_internal_RepeatedPtrFieldBase_Reserve)(int new_size);
+  onnx::TensorProto* (*google_protobuf_Arena_CreateMaybeMessage_onnx_TensorProto)(google::protobuf::Arena*);
+
+  // Special functions in bridge_special.h that route through these
+  virtual void onnx_AttributeProto_constructor(void* _this) = 0;
```

Comment:
>_this [](start = 53, length = 5)

All leading underscores are not per standard. Someone actually opened an issue on GH